packages/react-testing/src/compat.ts

```diff
@@ -34,3 +35,32 @@ export function createRoot(element: HTMLElement): ReactRoot {
     return createRootShim(element);
   }
 }
+
+export const isLegacyReact = parseInt(React.version, 10) < 18;
+
+// https://github.com/facebook/react/blob/12adaffef7105e2714f82651ea51936c563fe15c/packages/shared/enqueueTask.js#L13
+let enqueueTaskImpl: any = null;
+
+export function enqueueTask(task: (v: any) => void) {
```

Comment:
This is how the queue gets emptied out in older version of react. It creates a macro task either with `setImmediate` or `postMessage` . We're probably ok doing only the `setImmediate` here but decided to copy the entire code to be sure and match what react is doing.